ios/RNTPTDocumentView.m

```diff
@@ -4654,6 +4670,21 @@ - (NSString *) getDocumentPath {
     return [resultMap copy];
 }
 
+- (void)getFieldsForPage:(PTPage*)page resultMap:(NSMutableArray<NSDictionary *>*)resultMap
+{
+    int num_annots = [page GetNumAnnots];
+    for (int i = 0; i < num_annots; i ++){
+        PTAnnot *annot = [page GetAnnot:i];
+        if(annot != nil) {
+            if ([annot GetType] == e_ptWidget) {
+                __block NSDictionary* fieldMap = [self getFieldWithHasAppearance:annot];
+
+                [resultMap addObject:fieldMap];
+            }
+        }
+    }
+}
```

Comment:
I think it would be better to return the array itself instead of passing it in

```suggestion
- (NSArray<NSDictionary *>*)getFieldsForPage:(PTPage*)page
{
    NSMutableArray<NSDictionary *> *resultMap = [[NSMutableArray alloc] init];
    int num_annots = [page GetNumAnnots];
    for (int i = 0; i < num_annots; i ++){
        PTAnnot *annot = [page GetAnnot:i];
        if(annot != nil) {
            if ([annot GetType] == e_ptWidget) {
                __block NSDictionary* fieldMap = [self getFieldWithHasAppearance:annot];

                [resultMap addObject:fieldMap];
            }
        }
    }
    return [resultMap copy];
}
```